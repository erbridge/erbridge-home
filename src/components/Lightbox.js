import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Carousel, { Modal, ModalGateway } from 'react-images';

class Lightbox extends Component {
  static propTypes = {
    src: PropTypes.string.isRequired,
    isOpen: PropTypes.bool,
    onClose: PropTypes.func,
  };

  render() {
    const { src, isOpen, onClose } = this.props;

    return (
      <ModalGateway>
        {isOpen && (
          <Modal onClose={onClose} allowFullscreen={false}>
            <Carousel
              views={[{ src }]}
              components={{ FooterCount: () => <div /> }}
            />
          </Modal>
        )}
      </ModalGateway>
    );
  }
}

export default Lightbox;
