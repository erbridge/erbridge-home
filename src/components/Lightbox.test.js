import { shallow } from 'enzyme';
import React from 'react';

import Lightbox from './Lightbox';

it('renders correctly with all props', () => {
  expect(
    shallow(
      <Lightbox src="/test.jpg" isOpen onClose={() => console.log('test')} />,
    ),
  ).toMatchSnapshot();
});

it('renders correctly without optional props', () => {
  expect(shallow(<Lightbox src="/test.jpg" />)).toMatchSnapshot();
});

it('renders correctly when not open', () => {
  expect(
    shallow(<Lightbox src="/test.jpg" isOpen={false} />),
  ).toMatchSnapshot();
});
