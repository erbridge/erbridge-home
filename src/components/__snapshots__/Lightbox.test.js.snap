// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when not open 1`] = `
<ModalGateway
  target={<body />}
/>
`;

exports[`renders correctly with all props 1`] = `
<ModalGateway
  target={<body />}
>
  <Modal
    allowFullscreen={false}
    closeOnBackdropClick={true}
    closeOnEsc={true}
    onClose={[Function]}
    styles={Object {}}
  >
    <Carousel
      components={
        Object {
          "FooterCount": [Function],
        }
      }
      currentIndex={0}
      formatters={
        Object {
          "getAltText": [Function],
          "getCloseLabel": [Function],
          "getFullscreenLabel": [Function],
          "getNextLabel": [Function],
          "getNextTitle": [Function],
          "getPrevLabel": [Function],
          "getPrevTitle": [Function],
        }
      }
      hideControlsWhenIdle={3000}
      styles={Object {}}
      trackProps={
        Object {
          "instant": false,
          "swipe": "touch",
        }
      }
      views={
        Array [
          Object {
            "src": "/test.jpg",
          },
        ]
      }
    />
  </Modal>
</ModalGateway>
`;

exports[`renders correctly without optional props 1`] = `
<ModalGateway
  target={<body />}
/>
`;
